### PR TITLE
Implement simple `metadata()` for epoll and socket files

### PR DESCRIPTION
**Pull request by lucassong-mh**
Background: https://github.com/asterinas/asterinas/issues/1511

### CODE REVIEW

**Comment by tatetian** [comment]
What is Linux's `fstat` behaviors on epoll and socket FDs?

**Comment by lucassong-mh** [comment]
> What is Linux's `fstat` behaviors on epoll and socket FDs?

Elaberated in https://github.com/asterinas/asterinas/issues/1511:

The kernel panic in [CI job](https://github.com/asterinas/asterinas/actions/runs/11536668355/job/32113174947) and https://github.com/asterinas/asterinas/issues/1193 will be mitigated in https://github.com/asterinas/asterinas/pull/1520. (But may be only a short-term quick fix, elaberated below)

The unexpected behavior reveals two problems:

The first is `metadata()` in `trait FileLike`, that we should require this API forced to have each independent implementation. Beside that, means we need to implement `metadata()` for `EpollFile`/`UnixStreamSocket`/`StreamSocket`, etc for now. These special kind of files have their file discriptors, but not all of them have paths, or dentries, or inodes, and not all of them belong to a fs for now. So what's the behavior if we issue `fstat(fd)` to them?

What does Linux do? Well, for eventfd, eventpoll, timerfd, signalfd, inotifyfd, io_uring fd, Linux introduces "anonymous inode fs" [src link](https://github.com/torvalds/linux/blob/v6.12-rc5/fs/anon_inodes.c), which is a kernel in-memory fs that provides inodes and dentries for anonymous special files. The basic file creation, file_operations, inode number allocation, timestamps, exposed to `/proc`, etc, all goes to this special fs. So `fstat(fd)` is easy to follow, it just goes to the target inode maintained in `anon_inodefs`.
For socket files, similar, there is a "SockFS" [src link](https://github.com/torvalds/linux/blob/v6.12-rc5/net/socket.c), also a ram pseudo fs that maintains sockets/ports, provide posix compatible operations.

So the fix in https://github.com/asterinas/asterinas/pull/1520 is rather simple and dummy. If we're going to supprt these pseudo filesystems (anon inodefs, sockfs, pipefs, debugfs, etc). The filelike operations will be easier to have their rational implementations.


The second problem is about the [CI error log](https://github.com/asterinas/asterinas/actions/runs/11536668355/job/32113174947) . Here is a snippet.
```
Run Test Case: stat_test
Segmentation fault
Note: Google Test filter = -
[==========] Running 30 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 29 tests from StatTest

......

[----------] 1 test from SimpleStatTest
[ RUN      ] SimpleStatTest.AnonDeviceAllocatesUniqueInodesAcrossSaveRestore
[    32.013] ERROR: Uncaught panic: metadata unsupported
at /__w/asterinas/asterinas/kernel/src/fs/file_handle.rs:58
Printing stack trace:
   1: fn 0xffffffff[882](https://github.com/asterinas/asterinas/actions/runs/11536668355/job/32113174947#step:10:883)e83a0 - pc 0xffffffff882e842f / registers:
```
The `SimpleStatTest.AnonDeviceAllocatesUniqueInodesAcrossSaveRestore` resides in [test/syscall_test/blocklists/stat_test](https://github.com/asterinas/asterinas/blob/main/test/syscall_test/blocklists/stat_test)

Note that the panic occurs in `SimpleStatTest.AnonDeviceAllocatesUniqueInodesAcrossSaveRestore`, while, this case is supposed to be in the blocklist. And further look at the log we found that entering `stat_test` there occurs a `Segmentation fault`, and reading from the blocklist failed after that. So there exists a deeper bug in either memory or fs. I haven't had the time to dig yet.


References:
https://blog.csdn.net/FL63Zv9Zou86950w/article/details/120681279
https://stackoverflow.com/questions/4508998/what-is-an-anonymous-inode-in-linux
https://www.linux.org/threads/pipefs-sockfs-debugfs-and-securityfs.9638/

File: kernel/src/fs/epoll/epoll_file.rs [hunk 1]
```diff
@@ -341,7 +344,7 @@ impl Pollable for EpollFile {
     }
 }
 
-// Implement the common methods required by FileHandle
+// TODO: Think of introducing "anonymous inode fs" and link `EpollFile` to it.
```
Comment (tatetian) [inline kernel/src/fs/epoll/epoll_file.rs:347]: Could you move the comment close to the code, which is the metadata method?

```rust
    fn metadata(&self) -> Metadata {
        // This is a dummy implementation.
        // TODO: Add "anonymous inode fs" and link `EpollFile` to it.
        Metadata::new_file(
            0,
            InodeMode::from_bits_truncate(0o600),
            aster_block::BLOCK_SIZE,
        )
    }
```

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**